cmake/BLTMacros.cmake

```diff
@@ -586,7 +601,20 @@ macro(blt_add_executable)
         message(FATAL_ERROR "blt_add_executable(NAME ${arg_NAME} ...) given with no sources")
     endif()
 
-    add_executable( ${arg_NAME} ${arg_SOURCES} )
+    list(FIND arg_DEPENDS_ON "hip" check_for_hip)
```

Comment:
Can you pull all this hip logic into a separate macro in BLTPrivateMacros.cmake like blt_setup_cuda_target does? This macro is quite complicated and needs effort to stay readable.

Reply:
I've tried to re-locate any hip logic into separate blt private macros. Let me know if this change work for you.